client/js/components/header/search.js

```diff
@@ -0,0 +1,54 @@
+import showHide from './../show-hide';
+import { KEYS } from './../../util';
+
+const headerSearch = (el) => {
+  const headerSearchInput = el.querySelector('.js-header-input');
+  const searchToggle = showHide({
+    el: el,
+    activeClass: 'header__form--is-active'
+  });
+
+  const init = () => {
+    handleEvents();
+  };
+
+  const handleEvents = () => {
+    el.addEventListener('submit', (event) => {
+      if (searchToggle.getActive() && (headerSearchInput.value.trim() !== '')) return;
```

Comment:
Second thoughts, I think it should remain as is – better not to be able to submit the form when whatever text you might be searching for isn't visible on the page I think.